src/components/DateInput/DateInputPopover/index.jsx

```diff
@@ -0,0 +1,300 @@
+import React, { useRef, useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { Manager, Reference, Popper } from 'react-popper';
+import { CSSTransition } from 'react-transition-group';
+import FDS from 'lib/dictionary/js/styleConstants';
+import { isNotRefsEvent } from 'components/util/events';
+import { getPopperPlacement } from './util';
+import { useDisableScroll, useCloseOnScroll } from './hooks';
+
+export const VALID_POSITIONS = ['auto', 'top', 'right', 'bottom', 'left'];
+export const VALID_ALIGNMENTS = ['start', 'end', 'center'];
+export const VALID_INTERACTION_MODES = ['hover', 'click', 'controlled'];
+
+/**
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const Popover = React.forwardRef(
+  (
+    {
+      interactionMode = 'click',
+      disablePortal,
+      position = 'auto',
+      alignment = 'start',
+      distance = 4,
+      delay = 0,
+      onUserDismiss = () => {},
+      onOpen = () => {},
+      onClose = () => {},
+      trigger,
+      children,
+      isOpen,
+      transitionName,
+      disableScrollRef,
+      closeOnScrollRef,
+    },
+    forwardedRef
+  ) => {
+    const [isActive, setIsActive] = useState(false);
+    const refTriggerWrap = useRef(null);
+    const refContent = forwardedRef || React.createRef();
+
+    useCloseOnScroll(closeOnScrollRef, isActive, () => setIsActive(false));
+    useDisableScroll(disableScrollRef, isActive);
+
+    // update active state on props change to accommodate fully controlled popovers
+    useEffect(() => {
+      setIsActive(interactionMode === 'controlled' && isOpen);
+    }, [interactionMode, isOpen]);
+
+    /**
+     * Called when user takes an action to dismiss the popover
+     * @param {Event} e DOMEvent
+     */
+    const handleUserDismiss = (e) => {
+      onUserDismiss(e);
+      if (interactionMode !== 'controlled') {
```

Comment:
If `DateInput` is the only thing that uses this, do we need to support multiple interaction modes?

Reply:
I sort of don't see DateInputPopover as a component that will stick around. Me changing the name of the current Popover to DateInputPopover allows DateInput to still work, but the idea is that this component is on life support and would be trashed the MOMENT we get a chance, so I don't see it necessary to improve the component / remove cruft / removed unused props. That's just my perspective though, open to another viewpoint on it.

Reply:
"life support" works for me!